driver-core/src/main/java/com/datastax/driver/core/HostConnectionPool.java

```diff
@@ -396,7 +367,8 @@ private PoolingOptions options() {
           || maxTotalInFlight.compareAndSet(oldMax, totalInFlightCount)) break;
     }
 
-    int connectionCount = connections[shardId].size() + scheduledForCreation[shardId].get();
```

Comment:
Why did you move connections[shardId].size()? It's better to call it here to get the fresh value as other threads may be changing it.